src/adr/0040-quality-levels-definitions.md

```diff
@@ -0,0 +1,62 @@
+# 40. Quality levels definition
+
+Date: 2020-06-21
+
+## Status
+
+Accepted (lazy consensus)
+
+## Context
+
+We hereby define as an artifact compiled artifact that external people consumes. This includes:
+
+ - libraries
+ - Mail servers
+ - Extensions for James Mail Servers
+ - Command line tools
+
+We designate as a feature an optional, opt-in behaviour of a James server that can be configured by 
+user willing to rely on it.
+
+James as a project delivers several artifacts, and features. In order for project users to better
+understand the underlying quality of the artifact they use, as well as the level of risk associated,
+we need to better define some quality levels.
+
+## Decision
+
+For a given artifact or feature, by **mature** we mean that:
+
+ - *interfaces* in components need a contract test suite
```

Comment:
Would it be useful to describe "separation of API/implementation"? It is a term I have used a few times on the list, but I'm not sure I use it in the same way as others. (I have strongly corrupted by OSGi thinking, and it has a very clear meaning over there.)

"Interface" as you use it seems to suggest at the class level, whereas the real "interface" of a component is actualy at the package level, which may be why they call it "API" instead of "interface", to avoid confusion. Usually it is not just a single Java interface that defines behaviour, but rather an entire package. (Should generally not expand beyond a package.)

Reply:
> Would it be useful to describe "separation of API/implementation"?

:+1: and luckily it is the case almost everywhere.

> "Interface" as you use it seems to suggest at the class level, whereas the real "interface" of a component is actualy at the package level, which may be why they call it "API" instead of "interface", to avoid confusion. Usually it is not just a single Java interface that defines behaviour, but rather an entire package

If others don't complain, I will replace "interfaces" terms by "APIs".